internal/app/backend/backend_service.go

```diff
@@ -86,11 +113,23 @@ func (s *backendService) FetchMatches(req *pb.FetchMatchesRequest, stream pb.Bac
 		return err
 	}
 
+	results := proposals
```

Comment:
Might have meant to clone this proto.

Reply:
It is not redundant. We have proposals - and results. If evaluator is disabled case, the proposals assigned to results are the ones returned. If enabled, then the proposals variable is passed in where as the results are overwritten by the actual results - thus this variable is used to represent the results being passed out of the function.

Reply:
if evaluator enabled, its the result that gets updated with the actual results. if not, the proposals themselves are the results. hence the additional variable to store updated results if evaluator enabled. Again, once we move to enabling by default, this will vanish,

Reply:
Is it the same as the code below?
```
if s.synchronizerEnabled() {
		resp, err := s.synchronizer.EvaluateProposals(ctx, &internalpb.EvaluateProposalsRequest{
			Id:    syncID,
			Match: proposals})
		if err != nil {
			return err
		}

 		proposals = resp.Match
}
...
return doFetchMatchesSendResponse(ctx, proposals, sender)
```